internal/mode/static/metrics/collectors/controller.go

```diff
@@ -0,0 +1,58 @@
+package collectors
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/metrics"
+)
+
+// ControllerCollector collects metrics for the NGF controller.
+// Implements the prometheus.Collector interface.
+type ControllerCollector struct {
```

Comment:
I named it Controller because I figured we'd be adding more controller related metrics in the future. 

What if I named the interface that the handler depends on handlerMetricsCollector?

Reply:
> What if I named the interface that the handler depends on handlerMetricsCollector?

👍 

we do have controllers in the framework package, so I just worried about overloading the name controller
at the same time, we have the manager concept, because in manager.go, we start manager. But then the manager is also part of the controller-runitme...

I think controller is reasonable